Add FEATURE_IDEAS.md

TGWDB opened the pull request:
This adds FEATURE_IDEAS.md to gather together new features and research ideas
that otherwise would fill the issues/PR backlog. Also provide a simple
reference point for finding feature ideas or projects.

This now also combines the MINI_PROJECTS.md file into this one so there is a single location for all these kinds of ideas and projects.

Note that this would replace/close #28 #29 #40 #218 #265 #618  (and others as they are transitioned here).

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

TGWDB committed:
Add FEATURES.md

This adds FEATURES.md to gather together new features and research ideas
that otherwise would fill the issues/PR backlog. Also provide a simple
reference point for finding feature ideas or projects.

peterschrammel reviewed (approved):
I'd suggest to call the file `FEATURE_IDEAS.md` to make it clearer what its content is.

martin-cs reviewed (commented):
I would rather not do things this way.  There is already a mini-projects list:

https://github.com/diffblue/cbmc/pull/3930

which does get occasional updates:

https://github.com/diffblue/cbmc/pull/4119

It would be better to add to that than start a new file.  However despite being the original author of many of these I think that issues are a much better way of tracking this sort of thing.  They can be cross-referenced, have clearer updates from multiple people, integrate with the rest of how we work and have a much lower barrier to entry.

If this is a "number of open issues" issue then ... maybe a tag for "feature ideas" which can be excluded from the count?

martin-cs commented:
( The content is good though, as is making this sort of thing better recorded. )

TGWDB committed:
Combine distinct files into one

This removes the two files for FEATURES and MINI_PROJECTS and
combines them into a single FEATURE_IDEAS file.

TGWDB committed:
Add in another feature idea
